### PR TITLE
feat(gui): Use new rect type, delete GUIRect

### PR DESCRIPTION
**Pull request by eruvanos**

### CODE REVIEW

File: arcade/gui/property.py [hunk 1]
```diff
@@ -23,26 +23,26 @@ def __init__(self, value: P):
 
 class Property(Generic[P]):
     """
-    An observable property which triggers observers when changed.
+        An observable property which triggers observers when changed.
```
Comment (pushfoo) [inline arcade/gui/property.py]:
```suggestion
    """An observable property which triggers observers when changed.
```

File: arcade/gui/property.py [hunk 2]
```diff
@@ -23,26 +23,26 @@ def __init__(self, value: P):
...
-        # > Something changed
+            my_obj.name = "Hans"
+            # > Something changed
 
-    :param default: Default value which is returned, if no value set before
-    :param default_factory: A callable which returns the default value.
-                            Will be called with the property and the instance
+        :param default: Default value which is returned, if no value set before
+        :param default_factory: A callable which returns the default value.
+                                Will be called with the property and the instance
```
Comment (pushfoo) [inline arcade/gui/property.py]: This seems over-indented?

**Review by pushfoo — CHANGES_REQUESTED** [review]
TL;DR: I only have minor docstring complaints.
